File: Programing_Basics_with_JS/forLoopMoreExercises/02.Hospital.js
function hospital(input) {
    let index = 0;
    let period = Number(input[index]);
    index++;
    let numbersOfDoctors = 7;

    let examinedPatients = 0;
    let notExaminedPatients = 0;
    let workingDays = 1;

    for (let i = 1; i <= period; i++) {
        let numberOfPaтients = Number(input[index]);
        index++;
        let patientsDiff = numberOfPaтients - numbersOfDoctors;

        if (workingDays % 3 === 0 && notExaminedPatients > examinedPatients) {
            numbersOfDoctors++;
        }

        if (numbersOfDoctors >= numberOfPaтients) {
            examinedPatients += numberOfPaтients;
        } else {
            examinedPatients += numbersOfDoctors;
            notExaminedPatients += numberOfPaтients - numbersOfDoctors;
        }

        workingDays++;
    }

    console.log(`Treated patients: ${examinedPatients}.`);
    console.log(`Untreated patients: ${notExaminedPatients}.`);
}

hospital(['4', '7', '27', '9', '1'])
hospital(['6', '25', '25', '25', '25', '25', '2'])

File: Programming_Basics_with_Python/for_loop_more_exercises/02._hospital.py
days = int(input())

treated_patient = 0
untreated_patient = 0
doctors = 7

for day in range(1, days + 1):
    patients = int(input())

    if day % 3 == 0:
        if untreated_patient > treated_patient:
            doctors += 1

    if patients > doctors:
        untreated_patient += patients - doctors
        treated_patient += doctors
    else:
        treated_patient += patients

print(f'Treated patients: {treated_patient}.')
print(f'Untreated patients: {untreated_patient}.')



